### PR TITLE
draft of minimum viable programmer post

### PR DESCRIPTION
**Pull request by dbgrandi**

### CODE REVIEW

File: source/posts/minimum_viable_programmer.markdown [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+We were asked to describe our requirements for a *Minimum Viable Programmer*. These are what each of us  would expect from a junior developer. This turned out to be a fun exercise that brought out strong views from everybody.
+
+I wrote up a lot of notes for that meeting, and figured it would be a nice thing to share with all of you lovely people. It also seems to be a good bookend against my last post about [what makes for a great programmer](http://dbgrandi.github.io/good_great_10x/).
+
+## General skills a developer should have
+
+### | You can use simple git commands
+
+Every team I've worked with in the last 5 years has used `git`, with most using Github.
```
Comment (orta) [inline source/posts/minimum_viable_programmer.markdown]: GitHub

File: source/posts/minimum_viable_programmer.markdown [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+### | You can use simple git commands
+
+Every team I've worked with in the last 5 years has used `git`, with most using Github. 
+
+You should be able to add, commit, push, pull, branch, and merge. Knowing how to fork code, create a branch, and send a pull-request are vital to being able to contribute to a team.
+
+### | You know how to ask good questions
+
+Seriously. When a colleague comes to me with a question, my first response is almost always “What have you tried so far?" or “What did Google
```
Comment (orta) [inline source/posts/minimum_viable_programmer.markdown]: > or “What did Google

File: source/posts/minimum_viable_programmer.markdown [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+If you read books or blogs, attend meetups or conferences, and watch videos (WWDC videos!), then you are actively learning and growing. Software development is an immense field of study, and it is actively changing every day. There is always more to learn that will help you refine how you work.
+
+Some books that I'd recommend to any new programmer:
+
+- "The Pragmatic Programmer: From Journeyman to Master", by Andrew Hunt and David Thomas
+- "The UNIX Philosophy", by Mike Gancarz
+- "The Mythical Man-Month", by Fred Brooks
+- "The Cathedral and the Bazaar", by Eric S. Raymond
+- "The Design of Everyday Things", by Don Norman
```
Comment (orta) [inline source/posts/minimum_viable_programmer.markdown]: maybe have these link to amazon or goodreads?

**Comment by orta** [comment]
I feel like the ending could still have a bit more impact, the target reader for this is probably someone who may well be a MVP, you should find a way to inspire and give a call to action for something.

File: source/posts/minimum_viable_programmer.markdown [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+If it seems overwhelming, don’t lose hope. The more you learn about iOS the faster you can learn more. You start to see patterns that make it easier to understand a new API.
+
+And if you think that senior developers are immune from the frustrations that you have as a new programmer, here’s a few things that I did this year:
+
+- I’ve spent some number of hours tracking down misplaced semi-colons, commas, braces, or brackets.
+- I’ve thought that I understood how I should handle rotation in our app, only to realize I was completely wrong.
+- I’ve sworn that I found a critical-showstopper of a bug, only to have everything work fine when I deleted `DerivedData` and restarted Xcode.
+- I’ve sworn at my computer trying to figure out why Code Signing doesn’t work.
+- I’ve seen what to me what clearly the most simple and stupid error, and then checked the git history to find out that I wrote that code about a month ago.
```
Comment (alloy) [inline source/posts/minimum_viable_programmer.markdown]: > I’ve seen what to me ~~what~~ **was** clearly the most simple and stupid error, and then checked the git history to find out that I wrote that code about a month ago.

**Comment by alloy** [comment]
@dbgrandi Great post :clap: Gonna try to remember to show this to beginners when they ask these questions.

**Comment by dbgrandi** [comment]
/me tips :tophat: in your general direction.

Thanks @orta and @alloy for the feedback!

**Pull request merged**